[?1034hTrying:
    nl1 = Netlist()
Expecting nothing
ok
Trying:
    nl2 = Netlist()
Expecting nothing
ok
Trying:
    nl1.readYAML("test/gates.yml")
Expecting nothing
ok
Trying:
    nl2.readYAML("test/gates.yml")
Expecting nothing
ok
Trying:
    nl1.readVerilog("test/Iface_test.gv")
Expecting nothing
**********************************************************************
File "Netlist.py", line 35, in __main__.Netlist
Failed example:
    nl1.readVerilog("test/Iface_test.gv")
Exception raised:
    Traceback (most recent call last):
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.Netlist[4]>", line 1, in <module>
        nl1.readVerilog("test/Iface_test.gv")
      File "Netlist.py", line 272, in readVerilog
        mods = verilogParse.parseFile(verilogFile)
      File "/Users/apple/prj/eco/PyVerilog/verilogParse.py", line 732, in parseFile
        tokens = Verilog_BNF().parseFile( fileName )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 1410, in parseFile
        return self.parseString(file_contents, parseAll)
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 1021, in parseString
        loc, tokens = self._parse( instring, 0 )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2351, in parseImpl
        loc, resultlist = self.exprs[0]._parse( instring, loc, doActions, callPreParse=False )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2769, in parseImpl
        loc, tokens = self.expr._parse( instring, loc, doActions, callPreParse=False )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2478, in parseImpl
        ret = e._parse( instring, loc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2623, in parseImpl
        return self.expr._parse( instring, loc, doActions, callPreParse=False )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2368, in parseImpl
        loc, exprtokens = e._parse( instring, loc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2623, in parseImpl
        return self.expr._parse( instring, loc, doActions, callPreParse=False )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2742, in parseImpl
        loc, tmptokens = self.expr._parse( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2368, in parseImpl
        loc, exprtokens = e._parse( instring, loc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 894, in _parseNoCache
        loc,tokens = self.parseImpl( instring, preloc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 2478, in parseImpl
        ret = e._parse( instring, loc, doActions )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 952, in _parseCache
        value = self._parseNoCache( instring, loc, doActions, callPreParse )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 921, in _parseNoCache
        tokens = fn( instring, tokensStart, retTokens )
      File "/Library/Python/2.7/site-packages/pyparsing.py", line 675, in wrapper
        return func(*args[limit[0]:])
      File "/Users/apple/prj/eco/PyVerilog/verilogParse.py", line 202, in parseSubmod
        raise  Exception("Bad pin decl " + str(tmp))
    Exception: Bad pin decl ['ZN']
Trying:
    nl2.readYAML("test/Iface_test.yml")
Expecting nothing
ok
Trying:
    nl1.link("MYINVD1")
Expecting nothing
**********************************************************************
File "Netlist.py", line 37, in __main__.Netlist
Failed example:
    nl1.link("MYINVD1")
Exception raised:
    Traceback (most recent call last):
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.Netlist[6]>", line 1, in <module>
        nl1.link("MYINVD1")
      File "Netlist.py", line 71, in link
        " has not been defined"))
    Exception: link error, MYINVD1 has not been defined
Trying:
    nl1.topMod
Expecting:
    'MYINVD1'
**********************************************************************
File "Netlist.py", line 38, in __main__.Netlist
Failed example:
    nl1.topMod
Expected:
    'MYINVD1'
Got nothing
Trying:
    nl1.link("Iface_test")
Expecting nothing
**********************************************************************
File "Netlist.py", line 40, in __main__.Netlist
Failed example:
    nl1.link("Iface_test")
Exception raised:
    Traceback (most recent call last):
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.Netlist[8]>", line 1, in <module>
        nl1.link("Iface_test")
      File "Netlist.py", line 71, in link
        " has not been defined"))
    Exception: link error, Iface_test has not been defined
Trying:
    nl2.link("Iface_test")
Expecting nothing
ok
Trying:
    nl1.topMod
Expecting:
    'Iface_test'
**********************************************************************
File "Netlist.py", line 42, in __main__.Netlist
Failed example:
    nl1.topMod
Expected:
    'Iface_test'
Got nothing
Trying:
    nl2.topMod
Expecting:
    'Iface_test'
ok
Trying:
    mod1 = nl1.mods[nl1.topMod]
Expecting nothing
**********************************************************************
File "Netlist.py", line 46, in __main__.Netlist
Failed example:
    mod1 = nl1.mods[nl1.topMod]
Exception raised:
    Traceback (most recent call last):
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.Netlist[12]>", line 1, in <module>
        mod1 = nl1.mods[nl1.topMod]
    KeyError: None
Trying:
    mod2 = nl2.mods[nl2.topMod]
Expecting nothing
ok
Trying:
    set(mod1.ports.keys()) == set(mod2.ports.keys())
Expecting:
    True
**********************************************************************
File "Netlist.py", line 48, in __main__.Netlist
Failed example:
    set(mod1.ports.keys()) == set(mod2.ports.keys())
Exception raised:
    Traceback (most recent call last):
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.Netlist[14]>", line 1, in <module>
        set(mod1.ports.keys()) == set(mod2.ports.keys())
    NameError: name 'mod1' is not defined
Trying:
    set(mod1.cells.keys()) != set(mod2.cells.keys())
Expecting:
    True
**********************************************************************
File "Netlist.py", line 50, in __main__.Netlist
Failed example:
    set(mod1.cells.keys()) != set(mod2.cells.keys())
Exception raised:
    Traceback (most recent call last):
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.Netlist[15]>", line 1, in <module>
        set(mod1.cells.keys()) != set(mod2.cells.keys())
    NameError: name 'mod1' is not defined
Trying:
    set(mod1.nets.keys()) == set(mod2.nets.keys())
Expecting:
    True
**********************************************************************
File "Netlist.py", line 52, in __main__.Netlist
Failed example:
    set(mod1.nets.keys()) == set(mod2.nets.keys())
Exception raised:
    Traceback (most recent call last):
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.Netlist[16]>", line 1, in <module>
        set(mod1.nets.keys()) == set(mod2.nets.keys())
    NameError: name 'mod1' is not defined
Trying:
    nl1.checkDesign()
Expecting nothing
**********************************************************************
File "Netlist.py", line 54, in __main__.Netlist
Failed example:
    nl1.checkDesign()
Exception raised:
    Traceback (most recent call last):
      File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/doctest.py", line 1254, in __run
        compileflags, 1) in test.globs
      File "<doctest __main__.Netlist[17]>", line 1, in <module>
        nl1.checkDesign()
      File "Netlist.py", line 259, in checkDesign
        self.checkInputPorts()
      File "Netlist.py", line 120, in checkInputPorts
        print self.__mods[self.__topMod].ports.itesms()
    KeyError: None
18 items had no tests:
    __main__
    __main__.Netlist.__init__
    __main__.Netlist.__makePortTupleList__
    __main__.Netlist.addModule
    __main__.Netlist.checkConnectionDriver
    __main__.Netlist.checkConnectionWidth
    __main__.Netlist.checkDesign
    __main__.Netlist.checkInputPorts
    __main__.Netlist.checkOutputPorts
    __main__.Netlist.dumpPickle
    __main__.Netlist.findSubModules
    __main__.Netlist.link
    __main__.Netlist.mods
    __main__.Netlist.readVerilog
    __main__.Netlist.readYAML
    __main__.Netlist.topMod
    __main__.Netlist.writeVerilog
    __main__.Netlist.yaml
**********************************************************************
1 items had failures:
  10 of  18 in __main__.Netlist
18 tests in 19 items.
8 passed and 10 failed.
***Test Failed*** 10 failures.
